nautobot/extras/utils.py

```diff
@@ -87,3 +89,11 @@ def wrapper(model_class):
         return model_class
 
     return wrapper
+
+
+def generate_signature(request_body, secret):
+    """
+    Return a cryptographic signature that can be used to verify the authenticity of webhook data.
+    """
+    hmac_prep = hmac.new(key=secret.encode("utf8"), msg=request_body, digestmod=hashlib.sha512)
+    return hmac_prep.hexdigest()
```

Comment:
If you don't mind, I will open a housekeeping issue to track this under, rather than perverting this changeset. Just in case we break something, it will be easier to historically refer to the separate intent.